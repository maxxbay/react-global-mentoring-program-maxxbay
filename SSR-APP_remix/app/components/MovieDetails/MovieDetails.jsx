import React from 'react';
import { useLoaderData } from '@remix-run/react';
import PropTypes from 'prop-types';
import Button from '../Button/Button';
import { API_URL } from '../../constants';
import { Link } from 'react-router-dom';

import stylesDetails from '~/styles/MovieDetails.css';

export function MovieDetailsLinks() {
  return <link rel="stylesheet" href={stylesDetails} />;
}

export async function loader({ params }) {
  const movieId = params.movieId;
  console.log(movieId);
  const response = await fetch(`${API_URL}/${movieId}`);
  const movie = await response.json();
  return movie;
}

const MovieDetails = ({ movie, onClose }) => {
  const movie = useLoaderData();
  if (!movie) {
    return <p>Movie not found</p>;
  }
  const {
    poster_path: imageUrl,
    title: movieName,
    release_date: releaseYear,
    vote_average: rating,
    runtime: duration,
    overview: description,
  } = movie;

  const imageBaseUrl = 'https://image.tmdb.org/t/p/';
  const imageWidth = 'w185';

  const handleCloseClick = e => {
    e.stopPropagation();
    if (typeof onClose === 'function') {
      onClose();
    }
  };

  return (
    <div className="movie-details">
      <Link
        to={'/'}
        className="movie-details__close"
        onClick={handleCloseClick}
      >
        &times;
      </Link>
      <img
        className="movie-details__image"
        src={`${imageBaseUrl}${imageWidth}${imageUrl}`}
        alt={movieName}
      />
      <div className="movie-details__info">
        <h2 className="movie-details__title">{movieName}</h2>
        <p className="movie-details__year">{releaseYear}</p>
        <p className="movie-details__rating">Rating: {rating}</p>
        <p className="movie-details__duration">Duration: {duration} min</p>
        <p className="movie-details__description">{description}</p>
      </div>
    </div>
  );
};

export default MovieDetails;
