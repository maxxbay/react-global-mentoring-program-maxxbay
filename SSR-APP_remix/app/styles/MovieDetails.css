.movie-details-wrapper {
    padding: 1rem;
    background-color: #232323;
    color: #ffffff;
    width: 80%;
    display: flex;
    justify-content: center;
}

.movie-details {
    position: relative;
    display: flex;
    padding: 20px;
    align-items: center;
    justify-content: center;
   width: 90%;
}

.movie-details__image {
    width: 200px; 
    height: 300px; 
    object-fit: cover;
    border-radius: 5px;
    margin-right: 20px;
}

.movie-details__info {
    display: flex;
    flex-direction: column;
}

.movie-details__title {
    font-size: 30px;
    font-weight: 500;
    color: #fff;
    margin-bottom: 10px;
}

.movie-details__year {
    font-size: 20px; 
    color: #b8b3b3;
    margin-bottom: 10px;
}

.movie-details__rating,
.movie-details__duration {
    font-size: 18px; 
    color: #808080;
    margin-bottom: 10px;
}

.movie-details__description {
    font-size: 16px; 
    color: #808080;
    text-align: justify;
}

.movie-details__close {
   position: absolute;
    top: 50%;
    right: -20%;
    transform: translateY(-50%);
    background-color: #f65261;
    border: none;
    font-size: 24px;
    color: #fff;
    cursor: pointer;
    outline: none;
}

.movie-details__close:hover {
    background-color: #2e2425;
    color: #ffffff;
}
    
.movie-details__close:focus {
    box-shadow: 0 0 0 0.2rem #d87880;
}

