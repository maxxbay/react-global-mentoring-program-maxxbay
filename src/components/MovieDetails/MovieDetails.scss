.movie-details {
    position: relative;
    display: flex;
    padding: 20px;
    align-items: center;
    justify-content: center;
    width: 80%;

    &__image {
        width: 200px; 
        height: 300px; 
        object-fit: cover;
        border-radius: 5px;
        margin-right: 20px;
    }

    &__info {
        display: flex;
        flex-direction: column;
    }

    &__title {
        font-size: 30px;
        font-weight: 500;
        color: #fff;
        margin-bottom: 10px;
    }

    &__year {
        font-size: 20px; 
        color: #b8b3b3;
        margin-bottom: 10px;
    }

    &__rating,
    &__duration {
        font-size: 18px; 
        color: #808080;
        margin-bottom: 10px;
    }

    &__description {
        font-size: 16px; 
        color: #808080;
        text-align: justify;
    }

    &__close {
        position: absolute;
            top: 50%;
            right: -20%;
            transform: translateY(-50%);
            background-color: #f65261;
            border: none;
            font-size: 24px;
            color: #fff;
            cursor: pointer;
            outline: none;
    }


}
