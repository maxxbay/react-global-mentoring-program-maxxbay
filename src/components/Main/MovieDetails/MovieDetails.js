import React from 'react';
import './MovieDetails.scss';

const MovieDetails = ({ movie }) => {
  if (!movie) {
    return <div className="movie-details">Movie not found</div>;
  }

  const { poster_path, title, release_date, vote_average, runtime, overview } =
    movie;

  return (
    <div className="movie-details">
      <img
        className="movie-details__image"
        src={`https://image.tmdb.org/t/p/w185${poster_path}`}
        alt={title}
      />
      <div className="movie-details__info">
        <h2 className="movie-details__title">{title}</h2>
        <p className="movie-details__year">{release_date}</p>
        <p className="movie-details__rating">Rating: {vote_average}</p>
        <p className="movie-details__duration">Duration: {runtime} min</p>
        <p className="movie-details__description">{overview}</p>
      </div>
    </div>
  );
};

export default MovieDetails;
