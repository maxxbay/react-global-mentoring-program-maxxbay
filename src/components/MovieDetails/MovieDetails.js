import React from 'react';
import './MovieDetails.scss';

const MovieDetails = ({ movie }) => {
  if (!movie) {
    return <div className="movie-details">Movie not found</div>;
  }

  const {
    poster_path: imageUrl,
    title: movieName,
    release_date: releaseYear,
    vote_average: rating,
    runtime: duration,
    overview: description,
  } = movie;

  return (
    <div className="movie-details">
      <img
        className="movie-details__image"
        src={`https://image.tmdb.org/t/p/w185${imageUrl}`}
        alt={movieName}
      />
      <div className="movie-details__info">
        <h2 className="movie-details__title">{movieName}</h2>
        <p className="movie-details__year">{releaseYear}</p>
        <p className="movie-details__rating">Rating: {rating}</p>
        <p className="movie-details__duration">Duration: {duration} min</p>
        <p className="movie-details__description">{description}</p>
      </div>
    </div>
  );
};

export default MovieDetails;
