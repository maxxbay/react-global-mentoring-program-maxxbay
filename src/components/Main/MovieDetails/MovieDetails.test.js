import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import MovieDetails from './MovieDetails';

const mockMovie = {
  poster_path: '/mockPosterPath.jpg',
  title: 'Mock Movie Title',
  release_date: '2022-01-01',
  vote_average: 8.5,
  runtime: 120,
  overview: 'Mock movie overview',
};

describe('MovieDetails', () => {
  test('renders movie details correctly', () => {
    const { asFragment } = render(<MovieDetails movie={mockMovie} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders "Movie not found" when no movie is provided', () => {
    render(<MovieDetails />);
    expect(screen.getByText('Movie not found')).toBeInTheDocument();
  });

  test('renders movie information when provided with a movie', () => {
    render(<MovieDetails movie={mockMovie} />);
    expect(screen.getByText(mockMovie.title)).toBeInTheDocument();
    expect(screen.getByText(mockMovie.release_date)).toBeInTheDocument();
    expect(
      screen.getByText(`Rating: ${mockMovie.vote_average}`)
    ).toBeInTheDocument();
    expect(
      screen.getByText(`Duration: ${mockMovie.runtime} min`)
    ).toBeInTheDocument();
    expect(screen.getByText(mockMovie.overview)).toBeInTheDocument();
  });

  test('renders movie poster with correct src and alt attributes', () => {
    render(<MovieDetails movie={mockMovie} />);
    const moviePoster = screen.getByAltText(mockMovie.title);
    expect(moviePoster).toBeInTheDocument();
    expect(moviePoster).toHaveAttribute(
      'src',
      `https://image.tmdb.org/t/p/w185${mockMovie.poster_path}`
    );
  });
});
