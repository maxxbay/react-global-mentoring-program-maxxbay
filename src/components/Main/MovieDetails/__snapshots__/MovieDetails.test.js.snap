// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails renders movie details correctly 1`] = `
<DocumentFragment>
  <div
    class="movie-details"
  >
    <img
      alt="Mock Movie Title"
      class="movie-details__image"
      src="https://image.tmdb.org/t/p/w185/mockPosterPath.jpg"
    />
    <div
      class="movie-details__info"
    >
      <h2
        class="movie-details__title"
      >
        Mock Movie Title
      </h2>
      <p
        class="movie-details__year"
      >
        2022-01-01
      </p>
      <p
        class="movie-details__rating"
      >
        Rating: 8.5
      </p>
      <p
        class="movie-details__duration"
      >
        Duration: 120 min
      </p>
      <p
        class="movie-details__description"
      >
        Mock movie overview
      </p>
    </div>
  </div>
</DocumentFragment>
`;
